crates/fs/common/metadata.rs

```diff
@@ -69,7 +61,7 @@ pub struct UnixFsMetadata {
 }
 
 /// The metadata of a node on the WNFS file system.
-#[derive(Debug, Clone, PartialEq, Eq, FieldNames)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
Do we still need `derive(Deserialize)` on `Metadata`, `UnixFsMetadata` and `UnixFsMode` now that we implemented `TryFrom<&Ipld>` for them?

It may be confusing since there's two ways to serde-deserialize them then: serde -> Ipld -> Metadata and serde -> Metadata directly.

Reply:
We do actually. Ipld intermediate does not make them work with serde. Also it is used in PublicFile and PublicDirectory.